### PR TITLE
Added functionality to combine the indexes defined on the weave class with the target classes @Table annotation

### PR DESCRIPTION
**Pull request by cja769**
This was to alleviate a bug where having the Hibernate `@Index` on a column in an embeddable class resulted in a single index with all of the columns added by the weave instead of just adding an index for the specified columns in Hibernate 5+. The implementation of the merge was inspired by how we merge the `@EntityListeners` with some tweaks since the indexes are an array of annotations instead of an array of classes. With the change we should no longer add Hibernate `@Index` annotations to columns on embedded classes and instead add a JPA `@Table` annotation on the weave class. The `@Table` annotation won't cause any side-affects since the weave classes aren't hooked up to a persistence unit. However, if in the future we choose to register domain objects with class scanning then we'll have to simply create a new annotation for storing the indexes and update the `DirectCopyClassTransformer`.

This solves https://github.com/BroadleafCommerce/QA/issues/3408

### CODE REVIEW

File: common/src/main/java/org/broadleafcommerce/common/extensibility/jpa/copy/DirectCopyClassTransformer.java [hunk 1]
```diff
@@ -488,6 +491,114 @@ protected XFormParams reviewConditionalDirectCopyTransforms(String convertedClas
...
+                    for (Annotation annotation : items) {
+                        String typeName = annotation.getTypeName();
+                        if (typeName.equals(Table.class.getName())) {
+                            logger.debug("Stripping out previous Table annotation at the class level - will merge into new Table");
+                            existingTable = annotation;
+                            continue;
+                        }
+                        annotationsAttribute.addAnnotation(annotation);
+                    }
+                    itr.remove();
```
Comment (phillipuniverse) [inline common/src/main/java/org/broadleafcommerce/common/extensibility/jpa/copy/DirectCopyClassTransformer.java:531]: This appears to remove _all_ annotations from the `@Entity`. Seems like you should just be removing the `@Table`, right? What exactly are you removing here? Little difficult to follow (for good reason, this is difficult to work with).

Comment (cja769) [reply common/src/main/java/org/broadleafcommerce/common/extensibility/jpa/copy/DirectCopyClassTransformer.java:531]: @phillipuniverse An `AnnotationAttribute` is a basket that holds all of the annotations. That `itr.remove()` doesn't remove the `@Table` or `@Entity` annotation, it removes all class level annotations. However the `annotationsAttribute` list holds all of those annotation to be re-added later. This piece of code was copied from how we merge `@EntityListeners`

Comment (phillipuniverse) [reply common/src/main/java/org/broadleafcommerce/common/extensibility/jpa/copy/DirectCopyClassTransformer.java:531]: Got it. So it removes everything, but also keeps track of everything that it removed, except for the `@Table` annotation. That stays removed, but you hold onto it with the `existingTable` variable. Then you modify it below (or more technically, create a new one with your changes), and then add it to the `annotationsAttribute`, which gets added back as class-level annotations.

Comment (cja769) [reply common/src/main/java/org/broadleafcommerce/common/extensibility/jpa/copy/DirectCopyClassTransformer.java:531]: You got it dude!

**Pull request merged**